json11.cpp

```diff
@@ -435,7 +435,7 @@ struct JsonParser final {
     char get_next_token() {
         consume_garbage();
         if (i == str.size())
-            return fail("unexpected end of input", 0);
+            return (char)fail("unexpected end of input", 0);
```

Comment:
Given how the fail() template is structured, I think it would probably be better to cast the err_ret argument, rather than the return value.  I'd suggest changing the second argument to `(char)0` or '\0'.